src/gifwrite.c

```diff
@@ -241,7 +241,7 @@ void write_GIF_frame (struct context * context, const unsigned char * restrict d
 void write_GIF_data_blocks (struct context * context, const unsigned char * restrict data, size_t size) {
   uint8_t remainder = size % 0xff;
   size /= 0xff;
-  char * output = append_output_node(context, size * 0x100 + remainder + !!remainder + 1);
+  unsigned char * output = append_output_node(context, size * 0x100 + remainder + !!remainder + 1);
```

Comment:
This case of overflow is well defined — conversion by assignment always uses modulo arithmetic (as opposed to arithmetic overflow, which is truly UB for signed values). I'm not sure of why the compiler is warning for it... the change is harmless, but I don't understand the warning in the first place.

Reply:
Given that `data` is `unsigned char *` and it's copying from `data` to `output`, I'd make this change anyway so they match.

The warning was about the line `*(output ++) = 0xff`. The literal `0xff` is an `int`, so for signed `char`, which can't store 255, the signed-to-signed overflow is not defined (even if two's complement is by far the most common behavior).

Reply:
That conversion is well-defined because it's by assignment, as opposed to arithmetic overflow. Conversion by assignment is always modulo 2^N, which implies two's complement.
Still, the change is good :)

Reply:
I don't see anything about the conversion by assignment being well-defined in https://en.cppreference.com/w/c/language/conversion; it just says "if the target type is signed, the behavior is implementation-defined", and later "overflowing a signed integer type in an integer conversion is merely unspecified behavior". Maybe there's something about this in the C standard? Although you're wanting to support older versions and even C++, and it might not always be defined the same way?

Reply:
Hm, maybe I have it the wrong way around. I'll look into it.
At any rate, the change is good.